swap/htlc.go

```diff
@@ -210,78 +237,54 @@ func NewHtlc(version ScriptVersion, cltvExpiry int32,
 		builder.AddData(hash160)
 		builder.AddOp(txscript.OP_EQUAL)
 
-		pkScript, err = builder.Script()
+		pkScript, err := builder.Script()
 		if err != nil {
-			return nil, err
+			return nil, nil, nil, err
 		}
 
 		// Generate a valid sigScript that will allow us to spend the
 		// p2sh output. The sigScript will contain only a single push of
 		// the p2wsh witness program corresponding to the matching
 		// public key of this address.
-		sigScript, err = txscript.NewScriptBuilder().
+		sigScript, err := txscript.NewScriptBuilder().
 			AddData(p2wshPkScript).
 			Script()
 		if err != nil {
-			return nil, err
+			return nil, nil, nil, err
 		}
 
-		address, err = btcutil.NewAddressScriptHash(
+		address, err := btcutil.NewAddressScriptHash(
 			p2wshPkScript, chainParams,
 		)
 		if err != nil {
-			return nil, err
+			return nil, nil, nil, err
 		}
 
+		return address, pkScript, sigScript, nil
+
 	case HtlcP2WSH:
-		pkScript, err = input.WitnessScriptHash(htlc.Script())
+		pkScript, err := input.WitnessScriptHash(script)
 		if err != nil {
-			return nil, err
+			return nil, nil, nil, err
 		}
 
-		address, err = btcutil.NewAddressWitnessScriptHash(
+		address, err := btcutil.NewAddressWitnessScriptHash(
 			pkScript[2:],
 			chainParams,
 		)
 		if err != nil {
-			return nil, err
-		}
-
-	case HtlcP2TR:
-		// Confirm we have a v3 htlc.
-		trHtlc, ok := htlc.(*HtlcScriptV3)
-		if !ok {
-			return nil, ErrInvalidOutputSelected
-		}
-
-		// Generate a tapscript address from our HTLC's taptree.
-		address, err = btcutil.NewAddressTaproot(
-			schnorr.SerializePubKey(trHtlc.TaprootKey), chainParams,
-		)
-		if err != nil {
-			return nil, err
+			return nil, nil, nil, err
 		}
 
-		// Generate locking script.
-		pkScript, err = txscript.PayToAddrScript(address)
-		if err != nil {
-			return nil, err
-		}
+		// Pay to witness script hash (segwit v0) does not need a
+		// sigScript (we provide it in the witness instead), so we
+		// return nil for our sigScript.
+		return address, pkScript, nil, nil
 
 	default:
-		return nil, errors.New("unknown output type")
+		return nil, nil, nil, fmt.Errorf("unexpected output type: %v",
```

Comment:
Should we use `%d` here? Since `%v` will just render `unknown` if it's not a known type anyway, which doesn't really give us any more information. Or is this to make sure HtlcP2TR isn't used with `segwitV0LockingConditions`?